custom_components/tuya_local/devices/electriq_pd45e_dehumidifier.yaml

```diff
@@ -70,10 +70,37 @@ entities:
         name: lock
   - entity: sensor
     class: temperature
-    category: diagnostic
     dps:
       - id: 110
         type: integer
         name: sensor
         unit: C
         class: measurement
+  - entity: binary_sensor
+    class: problem
+    name: Tank full
```

Comment:
Use `translation_key: tank_full` here (no need for a class or icon either because name, icon and states are all overridden by the translation key). Also the icon override in the mapping is not needed.